ros2/source/devel/Dockerfile

```diff
@@ -0,0 +1,77 @@
+# This is an auto generated Dockerfile for ros2:devel
+# generated from docker_images_ros2/source/create_ros_image.Dockerfile.em
+FROM ubuntu:bionic
+
+# setup timezone
+RUN echo 'Etc/UTC' > /etc/timezone && \
+    ln -s /usr/share/zoneinfo/Etc/UTC /etc/localtime && \
+    apt-get update && apt-get install -q -y tzdata && rm -rf /var/lib/apt/lists/*
+
+# install packages
+RUN apt-get update && apt-get install -q -y \
+    bash-completion \
+    dirmngr \
+    gnupg2 \
+    lsb-release \
+    python3-pip \
+    wget \
+    && rm -rf /var/lib/apt/lists/*
+
+# setup keys
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116
+
+# setup sources.list
+RUN echo "deb http://packages.ros.org/ros2/ubuntu `lsb_release -sc` main" > /etc/apt/sources.list.d/ros2-latest.list
+
+# setup environment
+ENV LANG C.UTF-8
+ENV LC_ALL C.UTF-8
+
+# install bootstrap tools
+RUN apt-get update && apt-get install --no-install-recommends -y \
+    git \
+    python3-colcon-common-extensions \
+    python3-rosdep \
+    python3-vcstool \
+    && rm -rf /var/lib/apt/lists/*
+
+# install python packages
+RUN pip3 install -U \
+    argcomplete \
+    flake8 \
+    flake8-blind-except \
+    flake8-builtins \
+    flake8-class-newline \
+    flake8-comprehensions \
+    flake8-deprecated \
+    flake8-docstrings \
+    flake8-import-order \
+    flake8-quotes \
+    pytest-repeat \
+    pytest-rerunfailures
+
+ENV ROS_DISTRO crystal
```

Comment:
As we spoke offline, shall we move `ENV ROS_DISTRO` to source?

Reply:
I've been working on a refactor based on offline discussion, on this branchhttps://github.com/osrf/docker_templates/tree/ros2_devel_reboot, it will provide arg for ros distro in devel as well as args for testing ( and fail on test failure) in the source template. Not ready for review yet, I'll submit something in the coming days with the changes

Reply:
https://github.com/osrf/docker_templates/pull/57 now has a distro agnostic `devel` and allow users to set `ROS_DISTRO` on the source image using build arguments